packages/progress-bar/src/ProgressBar/ProgressBar.tsx

```diff
@@ -57,22 +57,31 @@ export function ProgressBar(props: ProgressBarProps) {
   );
 
   useEffect(() => {
-    if (animationMode === AnimationMode.Indeterminate && !isIndeterminate) {
-      setAnimationMode(AnimationMode.Transition);
-    }
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [isIndeterminate]);
+    setAnimationMode(currentMode => {
```

Comment:
I like this change! This way, you always have the most up-to-date animation mode.